cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/RegulatingControlMappingForTransformers.java

```diff
@@ -0,0 +1,391 @@
+        }
+        // order it is important
+        rtc.setRegulationTerminal(rca.terminal)
+            .setTargetDeadband(rca.targetDeadband)
+            .setTargetV(rca.targetValue);
+        rtc.setRegulating(rca.regulating);
+    }
+
+    private static void applyPhaseTapChanger(RegulatingControlPhaseAttributes rca, PhaseTapChanger ptc) {
+        if (rca == null) {
+            return;
+        }
+        // Order it is important
+        ptc.setRegulationTerminal(rca.terminal)
+            .setTargetDeadband(rca.targetDeadband)
+            .setRegulationValue(rca.targetValue)
+            .setRegulationMode(rca.regulationMode);
+        ptc.setRegulating(rca.regulating);
+    }
+
+    private void removeRatioControlIdFromCachedRegulatingControls(RegulatingControlRatioAttributes rcaRatio, String controlId) {
+        if (rcaRatio != null) {
+            parent.cachedRegulatingControls().remove(controlId);
+        }
+    }
+
+    private void removePhaseControlIdFromCachedRegulatingControls(RegulatingControlPhaseAttributes rcaPhase, String controlId) {
+        if (rcaPhase != null) {
+            parent.cachedRegulatingControls().remove(controlId);
+        }
+    }
+
+    private String getRegulatingControlId(PropertyBag p) {
+        String regulatingControlId = null;
+
+        if (p.containsKey(RegulatingControlMapping.TAP_CHANGER_CONTROL)) {
+            String controlId = p.getId(RegulatingControlMapping.TAP_CHANGER_CONTROL);
+            RegulatingControl control = parent.cachedRegulatingControls().get(controlId);
+            if (control != null) {
+                regulatingControlId = controlId;
+            }
+        }
+
+        return regulatingControlId;
+    }
+
+    private static boolean isControlModeVoltage(String controlMode, String tculControlMode) {
+        if (RegulatingControlMapping.isControlModeVoltage(controlMode)) {
+            return true;
+        }
+        return tculControlMode != null && tculControlMode.endsWith("volt");
+    }
+
+    private static boolean isControlModeFixed(String controlMode) {
+        return controlMode != null && controlMode.endsWith("fixed");
+    }
+
+    private static class RegulatingControlRatioAttributes {
+        Terminal terminal;
+        double targetValue;
+        double targetDeadband;
+        boolean regulating;
+    }
+
+    private static class RegulatingControlPhaseAttributes {
+        PhaseTapChanger.RegulationMode regulationMode;
+        Terminal terminal;
+        double targetValue;
+        double targetDeadband;
+        boolean regulating;
+    }
+
+    public static class RegulatingControlRatio extends RegulatingControlReference {
+        String id;
+        String tculControlMode;
+    }
+
+    public static class RegulatingControlPhase extends RegulatingControlReference {
+        boolean ltcFlag;
+    }
+
+    private static class RegulatingControlReference {
+        String regulatingControlId;
+        boolean tapChangerControlEnabled;
+    }
```

Comment:
I feel like there are too many transition classes to hold the data and this is too confusing. Could you only use `RegulatingControl`, a class named `RegulatingControlRatio` and a class named `RegulatingControlPhase`? There seems to be quite a lot of redundant information.

Reply:
i tried to clarify it selecting a more appropiate class name